homeassistant/components/tesla/sensor.py

```diff
@@ -17,89 +19,96 @@
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Tesla binary_sensors by config_entry."""
-    controller = hass.data[TESLA_DOMAIN][config_entry.entry_id]["controller"]
+    coordinator = hass.data[TESLA_DOMAIN][config_entry.entry_id]["coordinator"]
     entities = []
     for device in hass.data[TESLA_DOMAIN][config_entry.entry_id]["devices"]["sensor"]:
         if device.type == "temperature sensor":
-            entities.append(TeslaSensor(device, controller, config_entry, "inside"))
-            entities.append(TeslaSensor(device, controller, config_entry, "outside"))
+            entities.append(TeslaSensor(device, coordinator, "inside"))
+            entities.append(TeslaSensor(device, coordinator, "outside"))
         else:
-            entities.append(TeslaSensor(device, controller, config_entry))
+            entities.append(TeslaSensor(device, coordinator))
     async_add_entities(entities, True)
 
 
 class TeslaSensor(TeslaDevice, Entity):
     """Representation of Tesla sensors."""
 
-    def __init__(self, tesla_device, controller, config_entry, sensor_type=None):
+    def __init__(self, tesla_device, coordinator, sensor_type=None):
         """Initialize of the sensor."""
-        self.current_value = None
-        self.units = None
-        self.last_changed_time = None
+        super().__init__(tesla_device, coordinator)
         self.type = sensor_type
-        self._device_class = tesla_device.device_class
-        super().__init__(tesla_device, controller, config_entry)
 
-        if self.type:
-            self._name = f"{self.tesla_device.name} ({self.type})"
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return (
+            self.tesla_device.name
+            if not self.type
+            else f"{self.tesla_device.name} ({self.type})"
```

Comment:
We prefer to not use ternary operator for statements that span more than one line, for readability.

Reply:
@alandtse can you fix this in your followup PR? and ill merge it then. I would do if self.type return formatted string then default return of device name

Reply:
Done